### PR TITLE
New page for extend catalog rule conditions

### LINKED ISSUE #8224: add information on how to extend catalog price rule conditions
# New topic request
Could you please add information on how to extend catalog price rules to add custom conditions? 

## Description
Currently in the internet there's not much information about this funcionality but there are lots of questions regarding this subject. On the contrary extending Cart Rules has lots of documentation to find online. 

## Content checklist
It would be great to have an overlay of how to create a new rule condition and merge it with the default conditions. 

-  [ ] The topic provides an overview of how adding a condition works, what interfaces it extends and how it is applied.
-  [ ] The topic provides steps for building the needed classes and interfaces.
-  [ ] The topic contains code samples that shows how a core condition was implemented.

## Additional information/resources

Unfortunately information is very scarce on the web regarding this topic.

### PR DESCRIPTION
**Pull request by ajithkumar-maragathavel**
## Purpose of this pull request

This pull request (PR) adds a new page that explains the implementation of the catalog rule conditions and specifies the high-level steps to extend the conditions.

## Affected DevDocs pages

-  New Topic

## Links to Magento source code

-  https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/CatalogRule/etc/di.xml#L84
- https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/CatalogRule/Model/Rule/Condition/Combine.php
- https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/CatalogRule/Model/Rule/Condition/Product.php
- https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Rule/Model/Condition/Product/AbstractProduct.php

Fixes #8224 

whatsnew
Added the new [Extend Catalog Rule Conditions](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/extend-catalog-rule-conditions.html) topic.

### CODE REVIEW

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Great addition - just some grammar and language corrections.

File: src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+---
+title: Extend Catalog Rule Conditions
+contributor_name: Ziffity
+contributor_link: https://www.ziffity.com/
+group: php-developer-guide
+---
+
+Catalog price rules can be used to offer products to buyers at a discounted price, based on a set of defined conditions. Catalog price rules do not use coupon codes, because they are triggered before a product is placed into the shopping cart.
```
[the remark this answers is not in the document]
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md]: @BarnyShergold  I have redefined the second sentence. Please review it and let me know your suggestions.

**Comment by ajithkumar-maragathavel** [comment]
@BarnyShergold Thanks for the valuable suggestions. I have updated the PR. please review the changes.

File: src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+/**
+ * Validate product attribute value for condition
+ *
+ * @param \Magento\Catalog\Model\Product|\Magento\Framework\Model\AbstractModel $model
+ * @return bool
+ */
+public function validate(\Magento\Framework\Model\AbstractModel $model)
+{
+    $attrCode = $this->getAttribute();
+    if ('category_ids' == $attrCode) {
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md:93]: Sorry missed this one
```suggestion
    if ('category_ids' === $attrCode) {
```

**Review by BarnyShergold — APPROVED** [review]
one small change suggested, so I will approve and you can add the change - will be checked by the second reviewer

File: src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md (hunk 2 above)
Comment (dobooth) [reply src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md:93]: Good eye!

Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md]: @dobooth @BarnyShergold Updated the code. thanks for the review.

File: src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        <argument name="data" xsi:type="array">
+            <item name="form_name" xsi:type="string">catalog_rule_form</item>
+        </argument>
+    </arguments>
+</type>
+```
+
+In the above configuration, the `catalog_rule_form` is a UI component form. It is defined in the `app/code/Magento/CatalogRule/view/adminhtml/ui_component/catalog_rule_form.xml` file.
+
+The `Magento\CatalogRule\Model\Rule\Condition\Combine` and the `Magento\CatalogRule\Model\Rule\Condition\Product` classes are important for the listed conditions and validations of the specified conditions.
```
Comment (dobooth) [inline src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md:40]: Why are these important?

Comment (dobooth) [reply src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md]: I rewrote it. Please confirm if it is correct.

Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md]: @dobooth Yep, It is correct.

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import pr to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth the pull request successfully imported.

**Pull request merged**